common/tools/dev-tool/src/transpilers/samples.ts

```diff
@@ -0,0 +1,271 @@
+            log.debug(`File ${relativeSourcePath} has tag ${tag.tagName.text}`);
+
+            // New TS introduced comment: NodeArray, so we join the text if it is made of many nodes.
+            const comment = Array.isArray(tag.comment)
+              ? tag.comment.map((node: ts.JSDocText) => node.text ?? " ").join(" ")
+              : (tag.comment as string | undefined);
+
+            if (tag.tagName.text === "summary") {
+              log.debug("Found summary tag on node:", node.getText(sourceFile));
+              // Replace is required due to multi-line splitting messing with table formatting
+              summary = comment?.replace(/\s*\r?\n\s*/g, " ");
+            } else if (tag.tagName.text.startsWith(`${AZSDK_META_TAG_PREFIX}`)) {
+              // We ran into an `azsdk` directive in the metadata
+              const metaTag = tag.tagName.text.replace(
+                new RegExp(`^${AZSDK_META_TAG_PREFIX}`),
+                ""
+              ) as keyof AzSdkMetaTags;
+              log.debug(`File ${relativeSourcePath} has azsdk tag ${tag.tagName.text}`);
+              if (VALID_AZSDK_META_TAGS.includes(metaTag)) {
+                const trimmedComment = comment?.trim();
+                // If there was _no_ comment, then we can assume it is a boolean tag
+                // and so being specified at all is an indication that we should use
+                // `true`
+                azSdkTags[metaTag as keyof AzSdkMetaTags] = trimmedComment
+                  ? JSON.parse(trimmedComment)
+                  : true;
+              } else {
+                log.warn(
+                  `Invalid azsdk tag ${metaTag}. Valid tags include ${VALID_AZSDK_META_TAGS}`
+                );
+              }
+            }
+          }
+        }
+
+        return ts.visitEachChild(node, visitor, context);
+      };
+
+      ts.visitNode(sourceFile, visitor);
+
+      // We will only process exports for modules that appear to be util modules
+      if ((summary === undefined || azSdkTags.util) && exports.length > 0) {
+        return context.factory.updateSourceFile(sourceFile, [
+          ...sourceFile.statements,
+          // module.exports = { ... }
+          context.factory.createExpressionStatement(
+            context.factory.createAssignment(
+              context.factory.createPropertyAccessExpression(
+                context.factory.createIdentifier("module"),
+                context.factory.createIdentifier("exports")
+              ),
+              context.factory.createObjectLiteralExpression(
+                exports.map((name) => context.factory.createShorthandPropertyAssignment(name))
+              )
+            )
+          )
+        ]);
+      }
+
+      return sourceFile;
+    };
+
+    const jsModuleText = convert(sourceText, {
+      fileName: source,
+      transformers: {
+        before: [sourceProcessor]
+      }
+    });
+
+    if (summary === undefined && azSdkTags.util !== true) {
```

Comment:
I really don't like azsdk-util, and it only exists as a stopgap. What I'd like to do here is assume the file is a util file if it is imported by any other module, that way the default will be that the util mode/export analysis will be activated if it is needed, but can be overridden by the JSDoc comment.